pkg/k8s/apis/cilium.io/v2/types.go

```diff
@@ -458,3 +459,65 @@ func (n *CiliumNode) InstanceID() (instanceID string) {
 	}
 	return
 }
+
+// +genclient
+// +genclient:nonNamespaced
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// +kubebuilder:resource:singular="ciliumegressnatpolicy",path="ciliumegressnatpolicies",scope="Cluster"
+// +kubebuilder:printcolumn:JSONPath=".metadata.creationTimestamp",description="The age of the identity",name="Age",type=date
+// +kubebuilder:subresource:status
+// +kubebuilder:storageversion
+
+type CiliumEgressNATPolicy struct {
+	// +k8s:openapi-gen=false
+	// +deepequal-gen=false
+	metav1.TypeMeta `json:",inline"`
+	// +k8s:openapi-gen=false
+	// +deepequal-gen=false
+	metav1.ObjectMeta `json:"metadata"`
+
+	Spec CiliumEgressNATPolicySpec `json:"spec,omitempty"`
+}
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// +k8s:openapi-gen=false
+// +deepequal-gen=false
+
+// CiliumEgressNATPolicyList is a list of CiliumEgressNATPolicy objects.
+type CiliumEgressNATPolicyList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata"`
+
+	// Items is a list of CiliumEgressNATPolicy.
+	Items []CiliumEgressNATPolicy `json:"items"`
+}
+
+type CiliumEgressNATPolicySpec struct {
+	// Egress represents a list of rules by which egress traffic is
+	// filtered from the source pods.
+	Egress []EgressRule `json:"egress"`
+
+	// DestinationCIDRs is a list of destination CIDRs for destination IP addresses.
+	// If a destination IP matches any one CIDR, it will be selected.
+	DestinationCIDRs []string `json:"destinationCidrs"`
+
+	// EgressSourceIP is a source ip address that the egress traffice is
```

Comment:
```suggestion
	// EgressSourceIP is a source ip address that the egress traffic is
```

Reply:
@MasterZ40 You just need to fix that typo and we should be good to merge I think.

Reply:
fixed. Thanks!